Language_Parser/actionVerbs.py

```diff
@@ -325,17 +325,284 @@ def listen(info):
+        response = currentRoom.verbResponses("Use", item, feature)
+        # If that doesn't get a valid response, try swapping the inputs
+        if response == "None":
+            response = currentRoom.verbResponses("Use", feature, item)
+    except (KeyError, ValueError, IndexError):
+        response = None
+
+    if response != "None":
+        # Case for "catFood", "dish"
+        if {item, feature} == {"catFood", "dish"} and currentRoom == "porch":
+            # Remove catFood from inventory
+            removeOldItem(player, itemData[1], itemData[2])
+            # Call 'use' description
+
+            # Trigger condition
+            currentRoom.triggerCondition(feature, "Use")
+        else:
+            response = "That won't work."
+        return response
+    else:
+        return None
+
+
+def canOpenerCatFood(player, game, itemData2):
+    """
+    Helper function for 'combineTwoItems' -
+    applies the can opener item to the tin can to get cat food.
+    """
+    # Remove & consume tinCan
+    removeOldItem(player, itemData2[1], itemData2[2])
+
+    # Replace the tinCan with the catFood from Game storage
+    catFood = game.removeFromItemStorage("catFood")
+    placeNewItem(player, catFood, itemData2[2])
+
+
+def batteryFlashlight(player, game, itemData1, itemData2):
+    """
+    Helper function for 'combineTwoItems' -
+    applies a battery item to the flashlight item.
+    """
+    numFlashlights = 3
+
+    # Battery and previous flashlight are removed & consumed
+    removeOldItem(player, itemData1[1], itemData1[2])
+
+    # Remove existing flashlight from its location
+    removeOldItem(player, itemData2[1], itemData2[2])
+
+    # If the flashlight is still upgradable, get the upgrade name
+    if itemData2[3] < numFlashlights:
+        if itemData2[3] != 0:
+            upgradeIndex = str(itemData2[3] + 1)
+        else:
+            upgradeIndex = "2"
+
+        # Place the upgrade back where the old one was
+        upgrade = game.removeFromItemStorage(itemData2[0] + upgradeIndex)
+        placeNewItem(player, upgrade, itemData2[2])
+        game.removeFromItemStorage(upgrade.name)
+
+
+def combineTwoItems(player, game, item1, item2):
+    """
+    Helper function for 'useHandler'.
+    Takes player and game data, and two strings of item names.
+    Returns None if the combination fails.
+    The order of the item1 and item2 does not matter.
+    """
+    # Since order of words doesn't matter, choose an order for processing
+    if {item1, item2} == {"battery", "flashlight"}:
+        item1 = "battery"
+        item2 = "flashlight"
+        itemData1 = getItemDataForUse(player, item1)
+        itemData2 = getItemDataForUse(player, item2)
+    elif {item1, item2} in {"canOpener", "tinCan"}:
+        item1 = "canOpener"
+        item2 = "tinCan"
+        itemData1 = []  # canOpener doesn't need data
+        itemData2 = getItemDataForUse(player, item2)
+    else:
+        itemData1 = []
+        itemData2 = []
```

Comment:
If we know the scenario represented by this `else:` condition will eventually fail and return `None` below, perhaps it would be better to just return `None` here (rather than handling the empty lists below)?

In other words, rather than assigning empty lists here (which will raise an exception below and return `None`), we could just return `None` in the `else:` condition. Please correct me if I'm misunderstanding something. Thanks!

Reply:
Yeah that's probably fair, I'll look into changing this.